test/tools/unified-spec-runner/entities.ts

```diff
@@ -620,9 +620,8 @@ export class EntitiesMap<E = Entity> extends Map<string, E> {
               WriteConcern.fromOptions(defaultOptions);
           }
           if (defaultOptions.readPreference) {
-            options.defaultTransactionOptions.readPreference = ReadPreference.fromOptions(
-              defaultOptions.readPreference
-            );
+            options.defaultTransactionOptions.readPreference =
+              ReadPreference.fromOptions(defaultOptions);
```

Comment:
This isn't in my PR description, but we were originally parsing `defaultTransactionOptions` incorrectly in our entity creation in the UTR, and it needed to be fixed for the tests to pass.